tracing-tower/src/request_span.rs

```diff
@@ -0,0 +1,261 @@
+//! Middleware which instruments each request passing through a service with a new span.
+use super::GetSpan;
+use futures::{future::Future, Async, Poll};
+use std::marker::PhantomData;
+use tracing_futures::Instrument;
+
+#[derive(Debug)]
+pub struct Service<S, R, G = fn(&R) -> tracing::Span>
```

Comment:
do you think naming this `RequestService` might make it easier to debug service issues? aka when we get that giant stack error itll be easier to identify this instead of the trait.

Reply:
note that compiler errors will print fully-qualified type names, so this will be `tracing_tower::request_span::Service`. We could change the type name for the benefit of the `Debug` impl but I don't really see `Service`s printed at debug that often. If you think that would be better, though, I'm open to it for sure!

Reply:
I do agree with you, though I feel that overall in tower we have not been naming things just `Service` but `Retry` etc. So maybe there is a better name somewhere?

Reply:
I'm fine with either as long as we commit to a consistent style.